packages/scrollable/src/react/index.tsx

```diff
@@ -0,0 +1,206 @@
+}
+const Scrollable = forwardRef<HTMLElement, ScrollableProps>(
+  (props, forwardedRef) => {
+    const { renderContent = defaultRenderContent, ...rest } = props
+
+    const ref = useRef<HTMLElement>()
+    /* eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion */
+    useImperativeHandle(forwardedRef, () => ref.current as HTMLElement)
+
+    const dragPreview = useRef(BLANK_IMAGE)
+
+    const [hidden, setHidden] = useState<boolean>(true)
+    const [pageY, setPageY] = useState<number>(0)
+    const [offset, setOffset] = useState<string>('0%')
+    const [scrollRatio, setScrollRatio] = useState<number>(0)
+
+    const updateDimensions = useCallback(() => {
+      if (!ref.current) return
+      const content = ref.current
+
+      const nextTotalHeight = content.scrollHeight
+      const nextOffset = (content.scrollTop / nextTotalHeight) * 100 + '%'
+      const nextScrollRatio = content.clientHeight / nextTotalHeight
+      const shouldHide = nextScrollRatio >= 1
+
+      setHidden(shouldHide)
+      setOffset(nextOffset)
+      setScrollRatio(nextScrollRatio)
+    }, [ref])
+
+    useResizeObserver(ref, updateDimensions)
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    useEffect(() => updateDimensions(), [])
+
+    function onHandleDrag(evt: React.DragEvent<HTMLDivElement>) {
+      const nextPageY = evt.pageY
+      if (nextPageY <= 0) return
+
+      const content = ref.current
+      if (!content) return
+      const delta = evt.pageY - pageY
+
+      setPageY(nextPageY)
+      content.scrollTop += delta / scrollRatio
+    }
+
+    function onHandleDragEnd(_evt: React.DragEvent<HTMLDivElement>) {
+      setPageY(0)
+    }
+
+    function onHandleDragStart(evt: React.DragEvent<HTMLDivElement>) {
+      if (
+        evt &&
+        evt.dataTransfer &&
+        evt.dataTransfer.setDragImage &&
+        dragPreview.current
+      ) {
+        evt.dataTransfer.setDragImage(dragPreview.current, 0, 0)
+      }
+
+      setPageY(evt.pageY)
+    }
+
+    const styleOverride = useMemo(() => {
+      const height = Math.max(scrollRatio * 100, 10) + '%'
+      const visibility = hidden ? 'hidden' : 'initial'
+
+      return css({ height, top: offset, visibility })
+    }, [hidden, offset, scrollRatio])
+
+    return (
+      <Outer data-scrollable {...rest}>
+        <Inner key="scrollable-wrapper">
+          {renderContent(
+            {
+              children: props.children,
+              onScroll: updateDimensions,
+              ...styles.content()
+            },
+            (node: HTMLElement) => (ref.current = node)
+          )}
+        </Inner>
+
+        <Handle
+          draggable
+          onDrag={onHandleDrag}
+          onDragEnd={onHandleDragEnd}
+          onDragStart={onHandleDragStart}
+          {...styleOverride}
+        />
+      </Outer>
+    )
+  }
+)
+Scrollable.displayName = 'Scrollable'
+
+export interface renderContentProps {
+  /* [cssSelector: string]: string */
```

Comment:
remove?

Reply:
Yeah, probably.

I really wanted this line to be there, but I couldn't get it to jive well with tsc.  

I wanted it here because it indicates that the cssSelector is unknown in the value, but it will be present and  usable by anyone overriding `renderContent`.

But that overriding user won't know the value either, so I'm not sure just the instructions to "always spread the incoming props onto the content markup" aren't sufficient anyway.